server/controllers/finance/reports/financial.employee.js

```diff
@@ -42,22 +45,34 @@ function build(req, res, next) {
   }
 
   const data = {};
+  const sql = `
+    SELECT BUID(p.debtor_uuid) as debtor_uuid
+    FROM patient p, employee em
+    WHERE p.uuid = em.patient_uuid AND em.uuid = ?`;
 
-  return Employee.lookupEmployee(options.employee_uuid)
-    .then(employee => {
-      _.extend(data, { employee });
-      return Creditors.getFinancialActivity(employee.creditor_uuid);
-    })
-    .then(({ transactions, aggregates }) => {
-      aggregates.balanceText = aggregates.balance >= 0 ? 'FORM.LABELS.CREDIT_BALANCE' : 'FORM.LABELS.DEBIT_BALANCE';
-      _.extend(data, { transactions, aggregates });
-    })
-    .then(() => report.render(data))
-    .then(result => {
-      res.set(result.headers).send(result.report);
-    })
-    .catch(next)
-    .done();
+
+  // creditor information
+  // should addd something here
+  data.employee = await Employee.lookupEmployee(options.employee_uuid);
+
+  const { transactions, aggregates } = await Creditors.getFinancialActivity(data.employee.creditor_uuid);
+
+  aggregates.balanceText = aggregates.balance >= 0 ? 'FORM.LABELS.CREDIT_BALANCE' : 'FORM.LABELS.DEBIT_BALANCE';
+  _.extend(data, { transactions, aggregates });
+
+  // debtor informations
+  const patient = await db.one(sql, db.bid(options.employee_uuid));
```

Comment:
The use of async/await is really good!

We can make this more efficient by regrouping the queries and running a few of them in parallel with `Promise.all()`. Like this:

```js
const { employee, patient } = await Promise.all([
  Employee.lookupEmployee(options.employee_uuid),
  db.one(sql, db.bid(options.employee_uuid)),
]);

// now we have both the patient and employee
data.employee = employee;

// get debtor/creditor information
const [creditorOperations, debtorOperations] = await Promise.all([
  Creditors.getFinancialActivity(data.employee.creditor_uuid),
  Debitors.getFinancialActivity(patient.debtor_uuid, true),
]);

const { transactions, aggregates } = creditorOperations;
// etc...
```